gnssanalysis/gn_download.py

```diff
@@ -846,325 +848,169 @@ def download_multiple_files_from_cddis(files: [str], ftp_folder: str, output_fol
+) -> None:
     """
-    Function used to get the product file/s from download server of choice, default: CDDIS
-
-    Input:
-    dest - destination (str)
-    ac - Analysis Center / product of choice (e.g. igs, igr, cod, jpl, gfz, default = igs)
-    suff - optional suffix added to file name (e.g. _0 or _06 for ultra-rapid products)
-    f_type - file type to download (e.g. clk, cls, erp, sp3, sum, default = sp3)
-    dwn_src - Download Source (e.g. cddis, ga)
-    ftps - Optionally input active ftps connection object
-    wkly_file - optionally grab the weekly file rather than the daily
-    repro3 - option to download the REPRO3 version of the file
-
+    Download the file/s from CDDIS based on start and end epoch, to the
+    provided the download directory (download_dir)
     """
+    # DZ: Download the correct IGS FIN ERP files
+    if file_ext == "ERP" and analysis_center == "IGS" and solution_type == "FIN":  # get the correct start_epoch
+        start_epoch = GPSDate(str(start_epoch))
+        start_epoch = gpswkD2dt(f"{start_epoch.gpswk}0")
+        timespan = _datetime.timedelta(days=7)
+    # Details for debugging purposes:
+    logging.debug("Attempting CDDIS Product download/s")
+    logging.debug(f"Start Epoch - {start_epoch}")
+    logging.debug(f"End Epoch - {end_epoch}")
+
+    reference_start = _deepcopy(start_epoch)
+    product_filename, gps_date, reference_start = generate_product_filename(
+        reference_start,
+        file_ext,
+        long_filename=long_filename,
+        AC=analysis_center,
+        timespan=timespan,
+        solution_type=solution_type,
+        sampling_rate=sampling_rate,
+        project=project_type,
+    )
+    logging.debug(
+        f"Generated filename: {product_filename}, with GPS Date: {gps_date.gpswkD} and reference: {reference_start}"
+    )
+    with ftp_tls("gdc.cddis.eosdis.nasa.gov") as ftps:
+        try:
+            ftps.cwd(f"gnss/products/{gps_date.gpswk}")
+        except _ftplib.all_errors as e:
+            logging.info(f"{reference_start} too recent")
+            logging.info(f"ftp_lib error: {e}")
+            product_filename, gps_date, reference_start = generate_product_filename(
+                reference_start,
+                file_ext,
+                shift=-6,
+                long_filename=long_filename,
+                AC=analysis_center,
+                timespan=timespan,
+                solution_type=solution_type,
+                sampling_rate=sampling_rate,
+                project=project_type,
+            )
+            ftps.cwd(f"gnss/products/{gps_date.gpswk}")
+
+            all_files = ftps.nlst()
+            if not (product_filename in all_files):
+                logging.info(f"{product_filename} not in gnss/products/{gps_date.gpswk} - too recent")
+                raise FileNotFoundError
+
+        # reference_start will be changed in the first run through while loop below
+        reference_start -= _datetime.timedelta(hours=24)
```

Comment:
This acts like a shift interval. `reference_start` implies that it refers to the start epoch in some way.
```suggestion
        shift_interval -= _datetime.timedelta(hours=24)
```

Reply:
Yeah, so the `reference_start` is effectively a pointer used to determine where to start looking for files / downloading files.
This line shifts that `reference_start` back 24 hours, but the variable is returned: `-=` operation acts on the variable and returns it.
This is effectively: `reference_start = reference_start - _datetime.timedelta(hours=24)`

Reply:
Oops, not sure how I missed the `-` in that line.